src/solvers/lowering/byte_operators.cpp

```diff
@@ -1860,20 +1860,40 @@ static exprt lower_byte_update_struct(
         member.op0().swap(member.op1());
     }
 
-    byte_update_exprt bu{
-      src.id(),
-      std::move(member),
-      offset,
-      array_exprt{
-        std::move(update_values),
-        array_typet{bv_typet{8}, from_integer(update_size, size_type())}}};
+    exprt update_size_expr = from_integer(update_size, size_type());
+    array_exprt update_expr{std::move(update_values),
+                            array_typet{bv_typet{8}, update_size_expr}};
+    optionalt<exprt> member_update_bound;
+    if(non_const_update_bound.has_value())
+    {
+      const exprt remaining_update_bytes = typecast_exprt::conditional_cast(
```

Comment:
This construction is pretty difficult to follow; I'd appreciate a comment with pseudocode

Reply:
Comments added and code partly re-arranged to make things a bit easier to follow.